public/app/containers/Explore/PromQueryField.tsx

```diff
@@ -0,0 +1,282 @@
+      }
+    } else if (wrapperClasses.contains('context-labels') && !metricNode) {
+      // Empty name queries
+      const defaultKeys = ['job', 'instance'];
+      // Munge all keys that we have seen together
+      const labelKeys = Object.keys(this.state.labelKeys).reduce((acc, metric) => {
+        return acc.concat(this.state.labelKeys[metric].filter(key => acc.indexOf(key) === -1));
+      }, defaultKeys);
+      if ((text && text.startsWith('=')) || wrapperClasses.contains('attr-value')) {
+        // Label values
+        const labelKeyNode = getPreviousCousin(wrapperNode, '.attr-name');
+        if (labelKeyNode) {
+          const labelKey = labelKeyNode.textContent;
+          if (this.state.labelValues[EMPTY_METRIC]) {
+            const labelValues = this.state.labelValues[EMPTY_METRIC][labelKey];
+            context = 'context-label-values';
+            suggestions.push({
+              label: 'Label values',
+              items: labelValues.map(wrapText),
+            });
+          } else {
+            // Can only query label values for now (API to query keys is under development)
+            refresher = this.fetchLabelValues(labelKey);
+          }
+        }
+      } else {
+        // Label keys
+        context = 'context-labels';
+        suggestions.push({ label: 'Labels', items: labelKeys.map(wrapText) });
+      }
+    } else if (metricNode && wrapperClasses.contains('context-aggregation')) {
+      context = 'context-aggregation';
+      const metric = metricNode.textContent;
+      const labelKeys = this.state.labelKeys[metric];
+      if (labelKeys) {
+        suggestions.push({ label: 'Labels', items: labelKeys.map(wrapText) });
+      } else {
+        refresher = this.fetchMetricLabels(metric);
+      }
+    } else if (
+      (this.state.metrics && ((prefix && !wrapperClasses.contains('token')) || text.match(/[+\-*/^%]/))) ||
+      wrapperClasses.contains('context-function')
+    ) {
+      // Need prefix for metrics
+      context = 'context-metrics';
+      suggestions.push({
+        label: 'Metrics',
+        items: this.state.metrics.map(wrapText),
+      });
+    }
+
+    console.log('handleTypeahead', selection.anchorNode, wrapperClasses, text, offset, prefix, context);
+
+    return {
+      context,
+      prefix,
+      refresher,
+      suggestions,
+    };
+  };
+
+  async fetchLabelValues(key) {
+    const url = `/api/v1/label/${key}/values`;
+    try {
+      const res = await this.request(url);
+      const body = await (res.data || res.json());
+      const pairs = this.state.labelValues[EMPTY_METRIC];
+      const values = {
+        ...pairs,
+        [key]: body.data,
+      };
+      // const labelKeys = {
+      //   ...this.state.labelKeys,
+      //   [EMPTY_METRIC]: keys,
+      // };
+      const labelValues = {
+        ...this.state.labelValues,
+        [EMPTY_METRIC]: values,
+      };
+      this.setState({ labelValues });
+    } catch (e) {
```

Comment:
Is there a way we can centralize try/catch error handling more? Maybe via component decorators?  https://github.com/influxdata/chronograf/blob/master/ui/src/shared/decorators/errors.tsx

Reply:
Needs a better approach in general. I'd do this in a separate PR. For this one I harmonized the handling to simply `console.error()`